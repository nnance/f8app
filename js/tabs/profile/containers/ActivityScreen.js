import { connect } from 'react-redux';
import ActivityScreen from '../components/ActivityScreen';
import * as mockData from '../mockData';

export default connect(() => ({ activity: mockData.activity }))(ActivityScreen);
