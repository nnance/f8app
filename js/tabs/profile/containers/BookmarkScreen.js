import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import { updateMeBookmarksReduer, withRemoveBookmarks } from '../../../models/bookmark';

import BookmarkScreen from '../components/BookmarkScreen';

export const query = gql`
  query {
    me {
      id
      bookmarks {
        ...BookmarkScreen
      }
    }
  }
  ${BookmarkScreen.fragments.bookmark}
`;

const mapQueryToProps = ({ data }) => {
  if (data.error) {
    console.error(data.error);
  }
  return {
    bookmarks: !data.me || !data.me.bookmarks ? [] : data.me.bookmarks,
  };
};

export default withRemoveBookmarks(graphql(
  query,
  {
    props: mapQueryToProps,
    options: () => ({
      reducer: updateMeBookmarksReduer,
    }),
  },
)(BookmarkScreen));
