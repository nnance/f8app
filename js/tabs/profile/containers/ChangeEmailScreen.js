import React from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';

import F8Button from 'F8Button';
import { connect } from 'react-redux';

import SecureContainer from '../components/SecureContainer';
import NavBar from '../components/NavBar';

import { styles as commonStyles } from '../common';
import { colors as commonColors } from '../../../common/styles';

import { changeEmail } from '../../../actions/changeProfile';
import ModalSpinner from '../../../common/ModalSpinner';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: commonColors.nearlyWhiteBackground,
  },
  editorContainer: {
    backgroundColor: 'white',
    marginTop: 10,
  },
  button: {
    margin: 50,
  },
});

class ChangeEmailScreen extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      email: '',
      confirmEmail: '',
      error: null,
      saving: false,
    };
  }

  onChangeEmail() {
    const email = this.state.email;
    const confirmEmail = this.state.confirmEmail;
    this.setState({
      error: null,
      saving: true,
    });
    if (email !== confirmEmail) {
      this.setState({
        error: 'email not match',
        saving: false,
      });
      return Promise.resolve();
    }
    return this.props.changeEmail(email).then(() => {
      this.setState({
        svaing: false,
      });
      if (this.props.onBackPress) {
        this.props.onBackPress();
      }
    }).catch((error) => {
      this.setState({
        saving: false,
        error: error.message,
      });
    });
  }

  render() {
    return (<View style={styles.container}>
      <ModalSpinner visible={this.state.saving} />
      <NavBar
        title="เปลี่ยน Email"
        onBackPress={this.props.onBackPress}
      />
      <SecureContainer onCheck={() => Promise.resolve()}>
        <View>
          <View style={styles.editorContainer}>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>New Email</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="email"
                  autoCapitalize="none"
                  value={this.state.email || ''}
                  onChangeText={email => this.setState({ email })}
                />
              </View>
            </View>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>Confirm Email</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="confirm email"
                  autoCapitalize="none"
                  value={this.state.confirmEmail || ''}
                  onChangeText={confirmEmail => this.setState({ confirmEmail })}
                />
              </View>
            </View>
          </View>
          <View style={commonStyles.errorContainer}>
            <Text style={commonStyles.errorText}>{this.state.error}</Text>
          </View>
          <F8Button style={styles.button} caption="เปลี่ยน Email" onPress={() => this.onChangeEmail()} />
        </View>
      </SecureContainer>
    </View>);
  }
}

const actionsMaping = {
  changeEmail,
};

export default connect(null, actionsMaping)(ChangeEmailScreen);
export {
  ChangeEmailScreen as Component,
};
