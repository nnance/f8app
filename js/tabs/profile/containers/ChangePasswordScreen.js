import React from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
} from 'react-native';
import { connect } from 'react-redux';

import F8Button from 'F8Button';

import SecureContainer from '../components/SecureContainer';
import NavBar from '../components/NavBar';

import { styles as commonStyles } from '../common';
import { colors as commonColors } from '../../../common/styles';

import { changePassword } from '../../../actions/changeProfile';
import ModalSpinner from '../../../common/ModalSpinner';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: commonColors.nearlyWhiteBackground,
  },
  editorContainer: {
    backgroundColor: 'white',
    marginTop: 10,
  },
  button: {
    margin: 50,
  },
});

class ChangePasswordScreen extends React.Component {
  constructor(...args) {
    super(...args);
    this.state = {
      password: '',
      confirmPassword: '',
      error: null,
      saving: false,
    };
  }

  onChangePassword() {
    const password = this.state.password;
    const confirmPassword = this.state.confirmPassword;
    this.setState({
      error: null,
      saving: true,
    });
    if (password !== confirmPassword) {
      this.setState({
        error: 'password not match',
        saving: false,
      });
      return Promise.resolve();
    }
    return this.props.changePassword(password).then(() => {
      this.setState({
        saving: false,
      });
      if (this.props.onBackPress) {
        this.props.onBackPress();
      }
    }).catch((error) => {
      this.setState({
        saving: false,
        error: error.message,
      });
    });
  }

  render() {
    return (<View style={styles.container}>
      <ModalSpinner visible={this.state.saving} />
      <NavBar
        title="เปลี่ยน Password"
        onBackPress={this.props.onBackPress}
      />
      <SecureContainer onCheck={() => Promise.resolve()}>
        <View>
          <View style={styles.editorContainer}>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>New password</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="password"
                  autoCapitalize="none"
                  value={this.state.password || ''}
                  secureTextEntry
                  onChangeText={password => this.setState({ password })}
                />
              </View>
            </View>
            <View style={commonStyles.row}>
              <View style={commonStyles.label}>
                <Text>Confirm password</Text>
              </View>
              <View style={commonStyles.inputContainer}>
                <TextInput
                  style={{ flex: 1, height: 40 }}
                  placeholder="confirm password"
                  autoCapitalize="none"
                  value={this.state.confirmPassword || ''}
                  secureTextEntry
                  onChangeText={confirmPassword => this.setState({ confirmPassword })}
                />
              </View>
            </View>
          </View>
          <View style={commonStyles.errorContainer}>
            <Text style={commonStyles.errorText}>{this.state.error}</Text>
          </View>
          <F8Button style={styles.button} caption="เปลี่ยน Password" onPress={() => this.onChangePassword()} />
        </View>
      </SecureContainer>
    </View>);
  }
}

const actionsMaping = {
  changePassword,
};

export default connect(null, actionsMaping)(ChangePasswordScreen);
export {
  ChangePasswordScreen as Component,
};
