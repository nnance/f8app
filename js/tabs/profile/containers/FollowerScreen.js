import { connect } from 'react-redux';
import { FollowerScreen } from '../components/UserContainer';
import * as mockData from '../mockData';

export default connect(() => ({ userList: mockData.follower }))(FollowerScreen);
