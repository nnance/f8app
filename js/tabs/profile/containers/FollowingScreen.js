import { connect } from 'react-redux';
import { FollowingScreen } from '../components/UserContainer';
import * as mockData from '../mockData';

export default connect(() => ({ userList: mockData.following }))(FollowingScreen);
