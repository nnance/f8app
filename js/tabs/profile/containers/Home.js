import { connect } from 'react-redux';

import { logOutWithPrompt } from '../../../actions';

import Home from '../components/Home';

export default connect(
    state => ({
      user: state.user,
      followingCount: 1800,
      followerCount: 1800,
      candys: 100,
    }),
  {
    logOutWithPrompt,
  },
)(Home);
