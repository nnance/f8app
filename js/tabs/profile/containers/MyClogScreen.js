import { connect } from 'react-redux';
import MyClogScreen from '../components/MyClogScreen';
import * as mockData from '../mockData';

export default connect(() => ({ clogs: mockData.myClogs }))(MyClogScreen);
