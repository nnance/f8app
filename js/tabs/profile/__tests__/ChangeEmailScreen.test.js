import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import F8Button from 'F8Button';

import { Component as ChangeEmailScreenComponent } from '../containers/ChangeEmailScreen';

describe('ChangeEmailScreen', () => {
  function api(email) {
    if (email === 'fail@a.a') {
      return Promise.reject(new Error('something wrong'));
    }
    return Promise.resolve();
  }

  it('render authen', () => {
    const tree = renderer.create(<ChangeEmailScreenComponent />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('error if email not match', async () => {
    const wrapper = shallow(<ChangeEmailScreenComponent />);
    wrapper.find('[placeholder="email"]').simulate('changeText', 'a@a.a');
    wrapper.find('[placeholder="confirm email"]').simulate('changeText', 'b@a.a');
    wrapper.find(F8Button).simulate('press');
    expect(wrapper.state().error).toContain('not match');
  });

  it('call api', async () => {
    const changeEmail = jest.fn(api);
    const wrapper = shallow(<ChangeEmailScreenComponent changeEmail={changeEmail} />);
    wrapper.find('[placeholder="email"]').simulate('changeText', 'a@a.a');
    wrapper.find('[placeholder="confirm email"]').simulate('changeText', 'a@a.a');
    wrapper.find(F8Button).simulate('press');
    expect(changeEmail).toBeCalled();
  });

  it('call onBackPress if changed email', async () => {
    const changeEmail = jest.fn(api);
    const onBackPress = jest.fn();
    const wrapper = shallow(
      <ChangeEmailScreenComponent
        changeEmail={changeEmail}
        onBackPress={onBackPress}
      />,
    );
    wrapper.find('[placeholder="email"]').simulate('changeText', 'a@a.a');
    wrapper.find('[placeholder="confirm email"]').simulate('changeText', 'a@a.a');
    await wrapper.find(F8Button).props().onPress();
    expect(onBackPress).toBeCalled();
  });

  it('set state error if api reject', async () => {
    const changeEmail = jest.fn(api);
    const onBackPress = jest.fn();
    const wrapper = shallow(
      <ChangeEmailScreenComponent
        changeEmail={changeEmail}
        onBackPress={onBackPress}
      />,
    );
    wrapper.find('[placeholder="email"]').simulate('changeText', 'fail@a.a');
    wrapper.find('[placeholder="confirm email"]').simulate('changeText', 'fail@a.a');
    await wrapper.find(F8Button).props().onPress();
    expect(onBackPress).not.toBeCalled();
    expect(wrapper.state().error).toContain('something wrong');
  });
});
