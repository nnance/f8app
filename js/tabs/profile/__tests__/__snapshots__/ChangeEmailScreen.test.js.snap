exports[`ChangeEmailScreen render authen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(250, 250, 250)",
      "flex": 1,
    }
  }>
  <Modal
    transparent={true}
    visible={false}>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.6)",
          "flex": 1,
          "justifyContent": "center",
        }
      }>
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "height": 80,
            },
          ]
        } />
    </View>
  </Modal>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
          "paddingTop": 15,
          "width": 750,
        },
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#eeeeee",
        },
      ]
    }>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "flex": 1,
            "paddingLeft": 10,
          },
          undefined,
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
            }
          } />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 4,
          },
          undefined,
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 22,
              "fontWeight": "bold",
            },
            Object {
              "color": "rgb(249, 65, 92)",
              "fontSize": 22,
              "fontWeight": "bold",
            },
          ]
        }>
        เปลี่ยน Email
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "paddingRight": 10,
          },
          undefined,
        ]
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined} />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(250, 250, 250)",
          "flex": 1,
        },
        undefined,
      ]
    }>
    <View
      style={
        Object {
          "flex": 1,
        }
      } />
    <View
      style={
        Object {
          "flex": 8,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "red",
            "margin": 10,
          }
        } />
      <View
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.8)",
            "borderColor": "#eeeeee",
            "borderRadius": 10,
            "borderWidth": 0.5,
            "height": 80,
            "padding": 20,
            "width": 300,
          }
        }>
        <TextInput
          onChangeText={[Function]}
          placeholder="current password"
          secureTextEntry={true}
          style={
            Object {
              "flex": 1,
              "height": 40,
            }
          }
          value="" />
      </View>
      <View
        style={
          Object {
            "padding": 60,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 50,
              "opacity": 1,
            }
          }
          testID={undefined}>
          <BVLinearGradient
            colors={
              Array [
                4285164245,
                4285499097,
              ]
            }
            end={
              Object {
                "x": 1,
                "y": 1,
              }
            }
            locations={null}
            start={
              Object {
                "x": 0.5,
                "y": 1,
              }
            }
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 40,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 25,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": undefined,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "letterSpacing": 1,
                    },
                    Object {
                      "color": "white",
                    },
                  ],
                ]
              }>
              NEXT
            </Text>
          </BVLinearGradient>
        </View>
      </View>
    </View>
  </View>
</View>
`;
